Makefile

```diff
@@ -14,44 +18,45 @@ FFMPEG_OPTS ?= --prefix=$(FFMPEG_PREFIX) \
                --enable-parser=h264 \
                --enable-parser=hevc
 
-CGO_LDFLAGS := -L$(FFMPEG_PREFIX)/lib
+CGO_LDFLAGS := -L$(FFMPEG_BUILD)/lib
 ifeq ($(UNAME_S),Linux)
 	CGO_LDFLAGS := "$(CGO_LDFLAGS) -l:libjpeg.a"
 endif
 
-.PHONY: build-ffmpeg test lint updaterdk module clean
+.PHONY: build-ffmpeg lint update-rdk module clean clean-all
 
-bin/viamrtsp: build-ffmpeg *.go cmd/module/*.go
-	PKG_CONFIG_PATH=$(FFMPEG_PREFIX)/lib/pkgconfig \
+$(BIN_OUTPUT_PATH)/viamrtsp: build-ffmpeg *.go cmd/module/*.go
+	PKG_CONFIG_PATH=$(FFMPEG_BUILD)/lib/pkgconfig \
 		CGO_LDFLAGS=$(CGO_LDFLAGS) \
-		go build -o bin/viamrtsp cmd/module/cmd.go
-
-test:
-	go test
+		go build -o $(BIN_OUTPUT_PATH)/viamrtsp cmd/module/cmd.go
 
 lint:
 	gofmt -w -s .
 
-updaterdk:
+update-rdk:
 	go get go.viam.com/rdk@latest
 	go mod tidy
 
-FFmpeg:
-	git clone https://github.com/FFmpeg/FFmpeg.git --depth 1 --branch n6.1
+$(FFMPEG_VERSION_PLATFORM):
+	git clone https://github.com/FFmpeg/FFmpeg.git --depth 1 --branch $(FFMPEG_TAG) $(FFMPEG_VERSION_PLATFORM)
 
-$(FFMPEG_PREFIX): FFmpeg
-	cd FFmpeg && ./configure $(FFMPEG_OPTS) && $(MAKE) -j$(shell nproc) && $(MAKE) install
+$(FFMPEG_BUILD): $(FFMPEG_VERSION_PLATFORM)
```

Comment:
This will now have separate build directories per ffmpeg version & platform. I did this so that I can `make` from both `canon -arch amd64`, `canon -arch arm64` and from the OSX host & they can all safely build in parallel.

This dramatically sped up my ability to test different platforms

Reply:
```bash
ls FFmpeg/n6.1/
Darwin-arm64  Linux-aarch64 Linux-x86_64
```